src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -259,7 +264,7 @@ export function DiscoverMainRoute({
     async (nextDataView: unknown) => {
       if (nextDataView) {
         setLoading(true);
-        setShowNoDataPage(false);
+        setNoDataState({ showNoDataPage: false, hasESData: true, hasUserDataView: true });
```

Comment:
Or the following as it was before since we don't know if ES has data:

```suggestion
        setNoDataState(state => ({ ...state, showNoDataPage: false }));
```

Reply:
makes sense, thx